src/Microsoft.DotNet.Internal.Tools.SynchronizePackageProps/Program.cs

```diff
@@ -0,0 +1,397 @@
+                {
+                    if (propGroup.Attribute("Condition") != null)
+                    {
+                        WriteWarning($"{repoName} Packages.prop has Conditional PropertyGroup, which is not supported");
+                        continue;
+                    }
+
+                    foreach (XElement prop in propGroup.Elements())
+                    {
+                        if (prop.Attribute("Condition") != null)
+                        {
+                            WriteWarning(
+                                $"{repoName} Packages.prop property {prop.Name.LocalName} has Conditional PropertyGroup, which is not supported");
+                            continue;
+                        }
+
+                        props[prop.Name.LocalName] = prop.Value;
+                    }
+                }
+            }
+
+            var versions = new Dictionary<string, string>();
+            IEnumerable<XElement> packageReferences = packagesDocument?.Element("Project")
+                ?.Elements("ItemGroup")
+                ?.Elements("PackageReference");
+            if (packageReferences == null)
+            {
+                WriteError($"No PackageReferences found in {repoName} Packages.props");
+            }
+            else
+            {
+                foreach (XElement packageRef in packageReferences)
+                {
+                    string name = packageRef.Attribute("Update")?.Value;
+                    string version = packageRef.Attribute("Version")?.Value;
+
+                    if (string.IsNullOrEmpty(name))
+                    {
+                        WriteWarning($"PackageReference with no 'Update' found in Packages.props for repo {repoName}");
+                        continue;
+                    }
+
+                    if (string.IsNullOrEmpty(version))
+                    {
+                        WriteWarning(
+                            $"PackageReference with no 'Version' for Package '{name}' found in Packages.props for repo {repoName}");
+                        continue;
+                    }
+
+                    version = Regex.Replace(version,
+                        @"\$\(([A-Za-z0-9_]*)\)",
+                        m => props.GetValueOrDefault(m.Groups[1].Value, m.Groups[0].Value));
+
+                    versions.Add(name, version);
+                }
+            }
+
+            return versions;
+        }
+
+        private static void WriteError(string message)
+        {
+            Console.ForegroundColor = ConsoleColor.Red;
+            Console.Error.WriteLine(message);
+            Console.ResetColor();
+        }
+
+        private static void WriteWarning(string message)
+        {
+            Console.ForegroundColor = ConsoleColor.Yellow;
+            Console.Error.WriteLine(message);
+            Console.ResetColor();
+        }
+
+        [Flags]
+        private enum Errors
+        {
+            BadArguments = 1,
```

Comment:
Why is the Flags but not all the values are single bits? BadLocalFormat is BadArguments | MissingFile. Is that intentional?

Reply:
The first nibble is for "you called me wrong".  Once we've decided you called it right, we can detect a lot of errors at once, so figured it was helpful to just mush all those together. :-)